packages/core/src/runtime/StepIterator.ts

```diff
@@ -0,0 +1,156 @@
+import { ConditionFn, RecoverWith, Step, StepRecoveryObject } from './Step'
+import { Browser as BrowserInterface } from './IBrowser'
+import { Looper } from '../Looper'
+
+export default class StepIterator {
+	private steps: Step[]
+	private stepCount = 0
+	private currentStep: Step
+
+	constructor(allSteps: Step[]) {
+		this.steps = allSteps
+	}
+
+	get step(): Step {
+		return this.currentStep
+	}
+
+	goNextStep(): boolean {
+		this.stepCount += 1
+		return true
+	}
+
+	goPreviousStep(): boolean {
+		this.stepCount -= 1
+		return true
+	}
+
+	stepEnd() {
+		this.stepCount = this.steps.length
+		return true
+	}
```

Comment:
why these always return true? should these's return type be void instead?
```suggestion
	goNextStep(): void {
		this.stepCount += 1
	}

	goPreviousStep():void {
		this.stepCount -= 1
	}

	stepEnd():void {
		this.stepCount = this.steps.length
	}
```